### DISCUSSION TITLE
Fitting multiple data sets

### DISCUSSION THREAD

**Discussion by christianaz** [FrandsenGroup/beams#58] [Ideas]
This isn't necessarily a bug, just something I was wondering about. So I am trying to do multiple fits at once with a certain global variable. The problem is that I have some fixed variables that are different for each of the data sets. Is there a way that I can fix variables for the different data sets?

For example:
I want to use this equation: a0*(f+f2*(1-f)*exp(-λ1*t))
I want a0 to be the same fixed variable for all data sets. I want to set f as a global variable that is the same for all the data sets, but not a pre-determined number. I also want to fix alpha to a different number for each individual data set. How is the best way for me to be able to accomplish all of these things?

**Comment by aPeter1** [comment]
So this does not currently exist. It would be kind of weird to implement but in my mind you could have a couple different ways of doing it.

We could have a checkbox or something in the current table to specify this as an parameter to be specified individually. Then whenever a user selected a new run the table would change to show the one you have specified for that run. I feel like that table could be getting a bit crowded though.

We could have two separate tables of parameters. The first would be the table we have now, and for the second you could move variables down there and then you would have to specify a value specifically for each run (the table would change based on which run you currently have selected in the list below). 

@benfrandsen thoughts?

**Comment by aPeter1** [comment]
I really like this discussion feature btw, definitely should keep doing this 😄

**Comment by benfrandsen** [comment]
Here's my suggestion. The parameter table actually has three views called Config, Batch, and Output, with buttons at the top to allow the user to select which view they want. The image below shows roughly what I have in mind for each view.
![20210616_011409](https://user-images.githubusercontent.com/9492872/122174801-45404200-ce40-11eb-82b8-75ba2fc9485a.jpg)

The Config view is similar to what is currently in the table and should provide everything necessary for the user to do a fit to a single data set. The columns are Value, Min, Max, and Fixed, which should all be pretty self explanatory.

The Batch view allows the user to select which parameters should be treated as globally refined parameters in a batch fit, which should be fixed to a single value for all runs in the batch fit, and which should be fixed but with run-dependent values. The logic is as follows: For any given parameter, if "Fixed" is checkmarked on the Config page, then the checkbox for Global on the Batch page is unavailable, but instead the user has a checkbox for "Fixed, Run Dep.". If this checkbox is selected, then the user can enter text in the Value column to specify what it should be for each run. If the "Fixed, Run Dep." checkbox is not selected, then the "Value" column just reads in the fixed value assigned on the Config page and doesn't allow the user to change it. Finally, if the "Fixed" checkbox is unselected on the Config page for any given parameter, then the "Global" column on the Batch page has a checkbox available, but the "Fixed, Run. Dep." and "Value" columns are unavailable. If the Global checkbox is left unselected, then that particular parameter is just a run-dependent free parameter.

The Output view shows the results of the fit and consists of two columns: "Value" and "Uncertainty", where Value gives the final, best-fit value and Uncertainty gives the statistical uncertainty of each fitting parameter. To the side, we display the goodness-of-fit metrics chi-squared and reduced chi-squared for the specific run that has been selected and for the batch fit as a whole.

Does this make sense? I'm happy to talk you through it if it is unclear.

**Reply by aPeter1** [reply]
Makes sense to me, looks like a good way to get everything we need in one widget!

**Comment by christianaz** [comment]
Hello! How's it going with adding the new feature on to BEAMS?

**Reply by aPeter1** [reply]
Good, got the front end done, I'm just changing how I do things on the backend because I had some better ideas then what I did initially
